### ISSUE TITLE
Determining why a download fails to start

### ISSUE THREAD

**Issue by sehv** [bbilly1/tubearchivist#90]
I've noticed that certain videos won't download and end up stuck in the queue. So archiving a channel with 500 videos might end up with 10-20 videos that were skipped over and are now stuck in the queue. For some reason, the downloads never start, and manually starting the download also results in it sort of sitting on the Downloading: XYZ progress, but no data is actually being received. Killing the queue and retrying doesn't seem to affect it, nor does emptying the queue and re-adding the video.

Is there any way to tell why a video won't download? I can't seem to find any relevant info in any of the container's logs.

An example of this would be video ID: Mr5AdWTsosA
Doesn't appear to be any kind of age-gating or region locking or such, and I can fetch it through yt-dlp if I run it manually.

Slightly related, I'm a bit unclear about how the rescan functionality works. Suppose I've grabbed a channel with 500 videos in it, and have the above problem with 10 videos from across that full range of 500 not downloading. If I were to empty the queue and then rescan, will it only look at the first 50 videos from the channel and not add the older videos that haven't been fetched, or does it mean it will only add 50 videos at a time to the queue, but does in fact look across the entire channel's history?

**Comment by pairofcrocs** [comment]
I tried downloading the same video and the same thing is happening for me.

I actually have a couple hundred videos that are doing the exact same thing. I have ignored them for now, in hopes that future updates will fix said problem.

**Comment by sehv** [comment]
Doing a bunch of testing today, it turns out the issue with my videos has to do with missing chunks on youtube's side. When using the default format setting (whatever that happens to be), the videos exhibiting this behavior throw these sorts or errors when downloading through yt-dlp:
[download] Got server HTTP error: HTTP Error 404: Not Found. Retrying fragment 1 (attempt 1 of 10) ...

I found that by switching my format settings from whatever yt-dlp uses by default to something like -f best or -f mp4, it is able to find what it needs for that format. Not sure how that impacts overall quality, yt-dlp formats are definitely not something I can claim to really understand.

So my approach now is to let any new channel or rescan download with the default settings, then mop up whatever is leftover by changing the format to best or mp4. Probably not ideal, but it is getitng the job done.

**Comment by pairofcrocs** [comment]
Great find! 

Maybe @bbilly1can figure out a way to detect these, and use the default yt-dlp settings to give a download?

Nice investigate work!

**Comment by bbilly1** [comment]
Thanks for everybody looking into it. I observed the same, the video in question fails to download when not passing any format value with `-f`. Using `best` might not be the *best* solution (sorry, bad pun...) as this might not download the resolution you wanted to, `mp4` might not be ideal either, I don't think there are 4k streams in mp4 available, as far as I can tell, they are only in webm so you might be limited to 1080p only, which might or might not be a issue for you...

After a little bit of research I found `--check-formats` to let yt-dlp handle that instead. This is going to be a bit slower for every video, even if the first stream is available, but I guess, that's why you have a media server, an extra sec per video to validate the format is better than to end up failing the download. Then `--merge-output-format mp4` is already passed automatically to make it play in the browser.

There was a similar issue with the thumbnail extraction #83, same problem, yt-dlp detects a thumbnail that doesn't exist and same `--check-formats` fixes the issue but also with the same downside, adding to the download queue is going to take a bit more time to validate every thumbnail...
